drivers/counter/counter_smartbond_timer.c

```diff
@@ -47,15 +64,120 @@ struct counter_smartbond_config {
 	LOG_INSTANCE_PTR_DECLARE(log);
 };
 
+#if defined(CONFIG_PM_DEVICE) || defined(CONFIG_PM_DEVICE_RUNTIME)
+static void counter_smartbond_pm_policy_state_lock_get(struct counter_smartbond_data *data)
+{
+	if (atomic_test_and_set_bit(data->pm_policy_state_flag,
+				COUNTER_SMARTBOND_PM_POLICY_STATE_DRV) == 0) {
```

Comment:
atomic_test_and_set_bit returns bool so you should not use explicit comparison with int

Reply:
Fixed where needed